pyhf/modifiers/histosys.py

```diff
@@ -3,49 +3,68 @@
 
 from . import modifier
 from ..paramsets import constrained_by_normal
+from .. import get_backend, events
+from ..interpolate import _hfinterpolator_code0
 
-from .. import get_backend
-from ..interpolate import interpolator
-
-@modifier(name='histosys', constrained=True, shared=True, op_code = 'addition')
+@modifier(name='histosys', constrained=True, op_code = 'addition')
 class histosys(object):
-    def __init__(self, nom_data, modifier_data):
-        self.n_parameters = 1
-
-        self.at_zero = {}
-        self.at_minus_one = {}
-        self.at_plus_one = {}
-
+    @classmethod
+    def required_parset(cls, n_parameters):
+        n_parameters = 1
+        return {
+            'constraint': constrained_by_normal,
+            'n_parameters': n_parameters,
+            'modifier': cls.__name__,
+            'is_constrained': cls.is_constrained,
+            'is_shared': True,
+            'op_code': cls.op_code,
+            'inits': (0.0,) * n_parameters,
+            'bounds': ((-5., 5.),) * n_parameters,
+            'auxdata': (0.,) * n_parameters,
+            'factors': () * n_parameters
+        }
```

Comment:
now that all paramsets take `kwargs`, we should probably not even list these un-needed attributes here and elsewhere (.. unconstrained mods do not have auxdata.. only poisson constraints have factors